src/lib/FreeC/LiftedIR/Converter/Type.hs

```diff
@@ -13,43 +13,51 @@ module FreeC.LiftedIR.Converter.Type
 where
 
 import           Data.Bool                      ( bool )
-import           Data.Maybe                     ( fromJust )
 
 import qualified FreeC.IR.Syntax               as IR
 import           FreeC.IR.SrcSpan               ( SrcSpan(NoSrcSpan) )
 import qualified FreeC.LiftedIR.Syntax         as LIR
 
+import           FreeC.Monad.Converter          ( Converter )
+import           FreeC.Monad.Reporter           ( reportFatal
+                                                , Message(Message)
+                                                , Severity(Error)
+                                                )
+
 -- | Converts the argument types of a function.
 liftFuncArgTypes
   :: Maybe Int -- ^ Index of the decreasing argument for recursive functions.
   -> [IR.VarPat] -- ^ The argument types.
-  -> [LIR.Type]
+  -> Converter [LIR.Type]
 liftFuncArgTypes = maybe liftNonRecFuncArgTypes liftRecFuncArgTypes
 
 -- | Converts the argument types of a non recursive function using @convertType@.
-liftNonRecFuncArgTypes :: [IR.VarPat] -> [LIR.Type]
-liftNonRecFuncArgTypes = map $ fromJust . liftVarPatType
+liftNonRecFuncArgTypes :: [IR.VarPat] -> Converter [LIR.Type]
+liftNonRecFuncArgTypes = mapM $ \pat ->
+  let err = reportFatal $ Message (IR.varPatSrcSpan pat) Error "Expected Type"
```

Comment:
Since this error should never occur, I'd make it an `Internal` error.

```suggestion
  let err =
          reportFatal
            $ Message (IR.varPatSrcSpan pat) Internal
            $ "Expected variable pattern to have a type annotation."
```